internal/util/util.go

```diff
@@ -423,3 +428,27 @@ func Sample(m proto.Message, strategy *SamplingStrategy) proto.Message {
 	})
 	return m
 }
+
+// HashStore is a simple a store to produce hash for protobuf message
+type HashStore struct {
+	store map[string]int
+}
+
+// GetHash retrieves a hash string for a given protobuf message.
+// Note this should be restrict to a request scope.
+func (hs *HashStore) GetHash(m proto.Message) string {
+	if hs.store == nil {
+		hs.store = map[string]int{}
+	}
+	b, err := proto.Marshal(m)
+	if err != nil {
+		return "0"
```

Comment:
Should we return an error here instead of a fake 0?

Reply:
Use hash number now